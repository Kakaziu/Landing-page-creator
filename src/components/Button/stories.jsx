import Button from '.';

export default {
  title: 'Button',
  component: Button,
  args: {
    children: 'Clique',
  },
};

export const Template = (args) => (
  <div>
    <Button {...args} />
  </div>
);
