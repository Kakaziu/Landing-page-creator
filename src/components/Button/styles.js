import styled, { css } from 'styled-components';

const buttonSizes = {
  small: (theme) => css`
    padding: 8px 20px;
    font-size: ${theme.fonts.sizes.small};
  `,

  big: (theme) => css`
    padding: 8px 30px;
    font-size: ${theme.fonts.sizes.medium};
  `,
};

export const LinkContainer = styled.a``;

export const Button = styled.button`
  ${({ theme, size }) => css`
    font-weight: bold;
    border: none;
    background-color: ${theme.colors.secondaryColor};
    color: ${theme.colors.white};
    border-radius: 4px;
    ${buttonSizes[size](theme)}
  `}
`;
