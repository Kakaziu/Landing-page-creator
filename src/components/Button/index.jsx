import P from 'prop-types';
import * as Styled from './styles';

const Button = ({ children, size = 'small', url }) => {
  return (
    <Styled.LinkContainer href={url}>
      <Styled.Button size={size}>{children}</Styled.Button>
    </Styled.LinkContainer>
  );
};

Button.propTypes = {
  children: P.node.isRequired,
  size: P.oneOf(['small', 'big']),
  url: P.string,
};

export default Button;
