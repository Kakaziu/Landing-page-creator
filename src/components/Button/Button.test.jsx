import Button from '.';
import renderTheme from '../../styles/render-theme';

describe('<Button />', () => {
  it('should render button', () => {
    renderTheme(<Button>oi</Button>);
  });
});
