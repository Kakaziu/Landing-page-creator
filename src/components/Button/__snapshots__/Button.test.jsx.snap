// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
.c0 {
  display: block;
}

.c1 {
  font-weight: bold;
  border: none;
  background-color: #dc143c;
  color: #FFFFFF;
  border-radius: 4px;
  padding: 8px 20px;
  font-size: 1.6rem;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
  text-transform: uppercase;
}

.c1:hover {
  background-color: #b81132;
}

<div>
  <a
    class="c0"
    href="#"
  >
    <button
      class="c1"
    >
      Oi
    </button>
  </a>
</div>
`;
